torch/_dynamo/variables/torch.py

```diff
@@ -1092,16 +1007,39 @@ def speculate_branch(branch):
                 false_tensor_weakref_to_sizes_strides
             )
 
+            def fixup_branch_inps(graph, add_after, new_args, suffix) -> None:
+                inp_count = 0
+                for node in graph.nodes:
+                    if node.op == "placeholder":
+                        if inp_count == add_after:
+                            with graph.inserting_after(node):
+                                for inp_node in new_args:
+                                    new_node_name = inp_node.node.name + suffix
+                                    graph.placeholder(new_node_name)
+                            break
+                        inp_count += 1
+
+            fixup_branch_inps(
+                true_graph,
+                len(operands) + len(true_lifted_freevars) - 1,
+                false_lifted_freevars,
+                "_false_branch",
+            )
+
+            fixup_branch_inps(
+                false_graph, len(operands) - 1, true_lifted_freevars, "_true_branch"
+            )
+
             true_name = add_subgraph(
                 "cond_true",
-                torch.fx.GraphModule(true_nn_modules_context.nn_modules, true_graph),
+                torch.fx.GraphModule(true_nn_modules.nn_modules, true_graph),
             )
             false_name = add_subgraph(
                 "cond_false",
-                torch.fx.GraphModule(false_nn_modules_context.nn_modules, false_graph),
+                torch.fx.GraphModule(false_nn_modules.nn_modules, false_graph),
             )
 
-            # Apply side effects (guaranteed to be equal)
+            # TODO do we need this?
```

Comment:
We shouldn't, speculate_subgraph writes new side effects directly into tx.output.side_effects

Reply:
ok will delete